lib/rules/use-isnan.js

```diff
@@ -85,30 +85,31 @@ module.exports = {
                 /^(?:[<>]|[!=]=)=?$/u.test(node.operator) &&
                 (isNaNIdentifier(node.left) || isNaNIdentifier(node.right))
             ) {
-                context.report({
-                    node,
-                    messageId: "comparisonWithNaN",
-                    suggest: [
-                        {
-                            messageId: "replaceWithIsNaN",
-                            fix(fixer) {
-                                const comparedValue = isNaNIdentifier(node.left) ? node.right : node.left;
-                                let shouldNegate = ["!=", "!=="].includes(node.operator);
-
-                                if (comparedValue === node.right) {
-                                    shouldNegate ||= ["<", "<="].includes(node.operator);
-                                } else {
-                                    shouldNegate ||= [">", ">="].includes(node.operator);
+                const fixableOperators = ["==", "===", "!=", "!=="];
+                const isFixable = fixableOperators.includes(node.operator);
```

Comment:
Because this info never changes, you can move it outside of this method.

Also, it's more performant to use a `Set` rather than searching an array.

Reply:
Hmm
Is it faster in such small datasets?

[perf test](https://perf.link/#eyJpZCI6InBpeTV6c3R2NHRxIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6IiIsInRlc3RzIjpbeyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImNvbnN0IGEgPSBuZXcgU2V0KFsxLDIsMyw0XSk7XG5cbmEuaGFzKDQpIiwicnVucyI6WzgzNjAwMCw2NTQwMDAsMjc4OTAwMCwyMDUwMDAwLDE2NDYwMDAsMTI4NzAwMCwxMjg3MDAwLDYwMDAsMTI4NzAwMCwzMDUwMDAsMTY0NjAwMCwxNTU1MDAwLDUyNjAwMCwxMjg3MDAwLDUyMDAwLDkwMDAsMTI4NzAwMCwyMTI4MDAwLDE0MTYwMDAsMTE2MjAwMCw5OTUwMDAsMzE3NzAwMCwzMDAwMDAwLDIwNDEwMDAsNTc5NzAwMCw2MjAwMCwzMjI1MDAwLDE5NzMwMDAsMTI4NzAwMCwyMjgxMDAwLDk3MDAwMCwxNDcyMDAwLDk2NTAwMCw0NTYzMDAwLDI2OTQwMDAsOTgwMDAsMTI4NzAwMCw0NTYzMDAwLDEyODcwMDAsMjgyMzAwMCw1MDAwMCwyOTUwMDAwLDEyODcwMDAsODgzMDAwLDkxNTAwMCwxNjQ2MDAwLDUzNTAwMCw0Mjc5MDAwLDI4MDMwMDAsMTM5MDAwMCw5MjQwMDAsMTU5NTAwMCwxODY2MDAwLDY1MDAwLDM1MjMwMDAsMjE3OTAwMCw1ODAwMCwzOTY2MDAwLDIxOTcwMDAsMTI4NzAwMCw4NzIwMDAsMTI4NzAwMCwyMjU4MDAwLDMxNDAwMDAsMjgwNTAwMCwxMjg3MDAwLDEwOTAwMDAsMzcxMDAwLDQwMzMwMDAsMzI5OTAwMCw1ODkwMDAsMTY4MjAwMCwyMjM0MDAwLDE3NjcwMDAsNDU2MzAwMCwxNzAwMDAwLDM3NjcwMDAsMTI4NzAwMCwxNjgyMDAwLDMyMjUwMDAsMjczNzAwMCwzNjc2MDAwLDM0MTgwMDAsMTY0NjAwMCw3NTkwMDAsNDU2MzAwMCwzODUyMDAwLDUxMDAwMCw5NzkwMDAsMzA4OTAwMCwxNjM5MDAwLDE2NDYwMDAsMTI4NzAwMCwyNjg2MDAwLDc2NjAwMCw5MDAwMDAsMTI4NzAwMCwyNDAwMCwxODA0MDAwLDExNjEwMDBdLCJvcHMiOjE4MzU0MDB9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJjb25zdCBhID0gWzEsMiwzLDRdO1xuXG5hLmluY2x1ZGVzKDQpOyIsInJ1bnMiOlsxODE1MDAwLDIxNDQwMDAsNzkyODAwMCw3MTg3MDAwLDY0MzYwMDAsMjMyMDAwMCwyMTQ0MDAwLDQzNzAwMCwyMTQ0MDAwLDIzNzAwMCw2NTgxMDAwLDc2MDUwMDAsOTgwMDAsMjE0NDAwMCwxMTE3MTAwMCw5MDAwLDU5MDcwMDAsNTY5MDAwMCwyNjAxMDAwLDE2NDYwMDAsNTg0NDAwMCw4OTEyMDAwLDgzNDMwMDAsNTg5ODAwMCwxOTI0NDAwMCwxMTIxMDAwLDEwMDAwLDIwNzQwMDAsNDEzNTAwMCw3MDgyMDAwLDMyNjQwMDAsMjI3MzAwMCwyMzkyMDAwLDEzMzc1MDAwLDEzNzE2MDAwLDYwNjAwMCwyMTQ0MDAwLDk3ODkwMDAsNDAyNzAwMCwxMTYyMjAwMCw2MDkwMDAsMTMzODMwMDAsNTcwNTAwMCw5ODAwMCwxNTU3MDAwLDU1MjAwMDAsMTI3NTAwMCwxMDg0MDAwMCwyMTQ0MDAwLDYxMzAwMDAsMjE0NDAwMCwzNTgwMDAwLDU3MjgwMDAsMTI2MDUwMDAsOTAxNjAwMCw0OTI5MDAwLDAsMTE5MjEwMDAsMzQ3ODAwMCwyMjM1MDAwLDIxNDQwMDAsMjY1MDAwMCw2NzUyMDAwLDY4OTUwMDAsOTAwMCwyMTQ0MDAwLDMwMDYwMDAsMTM0MzMwMDAsOTcwMDAsMTAyMjgwMDAsMjE0NDAwMCw1MzExMDAwLDk4MDAwLDUyMTYwMDAsMTAwNjEwMDAsNjAxNjAwMCwxMTI5MjAwMCwyMjU5MDAwLDQ4MzMwMDAsODc1ODAwMCw2Njg1MDAwLDEwMjg4MDAwLDExMDUxMDAwLDUyMTgwMDAsMTIwNTAwMCwxNDM1MjAwMCw4ODY1MDAwLDI0MTUwMDAsMjIzNTAwMCw4OTY5MDAwLDUwMDEwMDAsNzI0OTAwMCwyMTQ0MDAwLDYxMzYwMDAsMjA0MTAwMCwxOTM1MDAwLDIxNDQwMDAsMTg4NTgwMDAsNDU3OTAwMCwzMDQzMDAwXSwib3BzIjo1MzI3NzEwfV0sInVwZGF0ZWQiOiIyMDI0LTAxLTE4VDA2OjA3OjEyLjAyNFoifQ%3D%3D)

Reply:
Microbenchmarks aren't a great indicator of real-world performance.

Reply:
Sure, done :)

It was a genuine question BTW
After thinking about it more, I think you're right, because if we move the Set creation outside the listener (and we did), it's being initialized only once (which is the "heavier" part AFAIK, correct me if I'm wrong), and then calling only `has()` is faster than `new Set()` + `has()`.
The benchmark I sent is creating the Set again each time the code runs, which hurts performance.

Anyway, thanks, TIL 😄